src/Umbraco.Core/Extensions/XmlExtensions.cs

```diff
@@ -234,6 +234,33 @@ public static XElement ToXElement(this XmlNode xmlElement)
             }
         }
 
+        public static T RequiredAttributeValue<T>(this XElement xml, string attributeName)
+        {
+            if (xml == null)
+            {
+                throw new ArgumentNullException("xml");
+            }
+
+            if (xml.HasAttributes == false)
+            {
+                throw new InvalidOperationException($"{attributeName} not found in xml");
+            }
+
+            if (xml.Attribute(attributeName) == null)
+            {
+                throw new InvalidOperationException($"{attributeName} not found in xml");
+            }
+
+            var val = xml.Attribute(attributeName).Value;
```

Comment:
Think we should do these check in one go (so we do not have `throw new InvalidOperationException($"{attributeName} not found in xml");` duplicated), and we cloud save `xml.Attribute(attributeName)` in a variable instead of executing it twice.